flask_praetorian/base.py

```diff
@@ -48,9 +63,10 @@ def __init__(self, app=None, user_class=None, is_blacklisted=None):
         self.pwd_ctx = None
         self.hash_scheme = None
         self.salt = None
+        self.app = app
 
         if app is not None and user_class is not None:
-            self.init_app(app, user_class, is_blacklisted)
+            self.app = self.init_app(app, user_class, is_blacklisted)
```

Comment:
flask extensions shouldn't necessarily be bound to an app instance. If the app object is needed, use `flask.current_app`

Reply:
this one was your code ;D

Reply:
I just mean, we should set `self.app` here

Reply:
Still super confused... it’s there, in green, yeah?

Reply:
I'm sorry for the confusion. My last comment was missing a critical 'not'.

We should not set self.app. If wee need the app object in praetorian code later, we can retrieve it in the flasky way with flask.current_app